core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -451,6 +451,15 @@ protected void doRun() {
                 }
                 performAction(node, transportPrimaryAction, true);
             } else {
+                if (state.version() < request.routedBasedOnClusterVersion()) {
+                    logger.trace("cluster state [{}] on node [{}] should have minimum version [{}] but was [{}], scheduling a retry: action [{}], request [{}]", request.shardId(), primary.currentNodeId(), request.routedBasedOnClusterVersion(), state.version(), actionName, request);
```

Comment:
maybe better to say  "failed to find primary despite of request being routing here. local cluster state version [{}]] is older than sending node (version [{}]), scheduling a retry... "